### PR TITLE
Fix invalid playlist ID crash

### PR DESCRIPTION
**Pull request by piazzatron**
### Description
- Fix undefined collection crash bug
- Better sentry logging

### Dragons

Nah

### How Has This Been Tested?

Locally

### CODE REVIEW

File: src/containers/collection-page/store/sagas.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ function* watchFetchCollection() {
 
     if (Object.values(collections).length === 0) {
       yield put(collectionActions.fetchCollectionFailed())
+      return
```
Comment (raymondjacobson) [inline src/containers/collection-page/store/sagas.js:23]: nice

File: src/utils/clientUtil.ts [hunk 2]
```diff
@@ -83,3 +85,9 @@ export const useIsMobile = () =>
   useMemo(() => {
     return isMobile()
   }, [])
+
+/**
+ * Whether we're deployed on our host or running locally
```
Comment (raymondjacobson) [inline src/utils/clientUtil.ts:90]: nit, or running elsewhere

File: src/utils/clientUtil.ts [hunk 3]
```diff
@@ -48,7 +48,8 @@ export const getOS = () => {
 }
 
 export const getMobileOS = () => {
-  const userAgent = navigator.userAgent || navigator.vendor || window.opera
+  const userAgent =
+    navigator.userAgent || navigator.vendor || (window as any).opera
```
Comment (raymondjacobson) [inline src/utils/clientUtil.ts:52]: nit, at the top of the file you can do

```
declare global {
  interface Window {
    opera: any
  }
}
```

**Review by raymondjacobson — APPROVED** [review]

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/aud-135

File: src/utils/clientUtil.ts (hunk 3 above)
Comment (piazzatron) [reply src/utils/clientUtil.ts:52]: oh much better

**Pull request merged**